chaco/tools/range_selection.py

```diff
@@ -679,6 +679,14 @@ def _determine_axis(self):
     def __mapper_changed(self, event):
         self.deselect()
 
+    def _component_changed(self, old, new):
+        if old is not None:
+            self.plot.observe(
+                self.__mapper_changed, self.axis + "_mapper", remove=True
+            )
+        if new is not None:
+            self.plot.observe(self.__mapper_changed, self.axis + "_mapper")
+
```

Comment:
semi-relevant (?) ref: https://github.com/enthought/enable/issues/739
See this class' base class, `AbstractController`'s `__init__`method:
https://github.com/enthought/chaco/blob/3e83091a8529cfde347c99df35c131b2df479d7e/chaco/abstract_controller.py#L16-L18

Reply:
Probably. Dynamic trait change handlers (`@on_trait_change()` and `@observe()`) need `HasTraits.__init__()` to run before the machinery is properly set up. Static trait change handlers like `_component_changed` don't need that machinery to be set up and will be run on the relevant attribute setting.

Reply:
ah that is very good to know, thank you!